src/Data/String.purs

```diff
@@ -199,6 +207,7 @@ foreign import _lastIndexOf'
 foreign import length :: String -> Int
 
 -- | Locale-aware sort order comparison.
+-- | - `"ä" `localeCompare` "b" == LT`
```

Comment:
Perhaps we could contrast this with the `Ord` instance which treats strings as arrays of code units here?